DOC-14357: Document cbcat tool (#4133)
couchbase/docs-server: modules/cli/pages/cbcat.adoc

= cbcat
:description: pass:q[The `cbcat` tool decrypts and displays the contents of encrypted logs and archive files.]
:totoro: the next major release of Couchbase Server
:page-status: {totoro}
:page-topic-type: reference

[abstract]
{description}

== Syntax

The basic syntax is:

----
cbcat [options] file(s)
----

where `file(s)` is a file or multiple files to decrypt and view.
You can use any method supported by the shell to specify multiple files, including wildcards.

== Description

> This tool is available in {totoro} and later.

Reviewer: It is also present in 8.0

When encryption at rest is active, Couchbase Server encrypts sensitive data stored on disk.
This includes log files and archive files generated by Couchbase Server, and by services that are running in the cluster.
The `cbcat` tool enables you to decrypt and view the contents of encrypted logs and archive files that were created when encryption at rest was active.

Depending upon your platform, this tool is at the following locations:

[cols="1,3"]
|===
| Operating system | Location

| Linux
| `/opt/couchbase/bin/`

| Windows
| `C:\Program Files\Couchbase\Server\bin\`

| Mac OS X
| `/Applications/CouchbaseServer.app/Contents/Resources/couchbase-core/bin/`
|===

== Options

--with-dump-keys <filename>::
The name and location of the `dump-keys` binary.
This binary connects to the keystore and gets the encryption keys.

> If not specified, the default is `/opt/couchbase/bin/dump-keys`.

Reviewer: The default in totoro would be the same directory as cbcat

+
You do not usually need to use this option if you installed Couchbase Server in the default location.

--with-gosecrets <filename>::
The name and location of the `gosecrets.cfg` configuration file.
This file contains information about the location of encryption keys, and other encryption-at-rest settings.

> If not specified, the default is `/opt/couchbase/var/lib/couchbase/config/gosecrets.cfg`.

Reviewer: THe default in totoro would be relative to the installation of cbcat (../var/lib etc)

= cbcat
:description: pass:q[The `cbcat` tool decrypts and displays the contents of encrypted logs and archive files.]
:page-topic-type: reference

[abstract]
{description}

== Syntax

The basic syntax is:

----
cbcat [options] file(s)
----

where `file(s)` is a file or multiple files to decrypt and view.
You can use any method supported by the shell to specify multiple files, including wildcards.

== Description

When encryption at rest is active, Couchbase Server encrypts sensitive data stored on disk.
This includes logs, configuration files, audit files, and other files generated by Couchbase Server and the services running in the cluster.

> The `cbcat` tool enables you to decrypt and view the contents of encrypted files that were created when encryption at rest was active.

Reviewer 2: ```suggestion
The `cbcat` tool enables you to decrypt and view the contents of these encrypted files.
```
I think we don’t need to repeat “when encryption at rest was active” since that’s already mentioned in the first line.

Author: Done

| Mac OS X
| `/Applications/CouchbaseServer.app/Contents/Resources/couchbase-core/bin/`
|===

== Options

--with-dump-keys <filename>::
The name and location of the `dump-keys` binary.
This binary connects to the keystore and gets the encryption keys.
If not specified, the default is `./dump-keys`, in the same location where `cbcat` is installed.
+
You do not usually need to use this option if you installed Couchbase Server in the default location.

--with-gosecrets <filename>::
The name and location of the `gosecrets.cfg` configuration file.
This file contains information about the location of encryption keys, and other encryption-at-rest settings.
If not specified, the default is `../var/lib/couchbase/config/gosecrets.cfg`, relative to the location where `cbcat` is installed.
+
You do not usually need to use this option if you installed Couchbase Server in the default location.

--password <password>::
The master password to use for decryption.
Use `-` to prompt for a password or to read from standard input.
+
This option and the `--with-keystore` option are mutually exclusive.
You can only use one of these options.

--with-keystore <json>::

> A JSON representation of a keystore containing the keys that were used to encrypt the files you want to decrypt and view.

Reviewer 2: Maybe add a link to the Schema section?

Author: Done

    Key Derivation: KeyBased
    Compression: None
======================================================================
<file content ...>
----
====

[[cbcat-ex6]]
.Decrypt and view file using its encryption key
====
The following command finds the value of a data encryption key, using the key ID returned by <<cbcat-ex5>>.

.Command
[source,sh]
----
dump-keys -p $MASTER -k 3f64f34c-a0bd-4248-a1da-5769e5bc15ef
----

.Output
[source,json]
----
{
  "3f64f34c-a0bd-4248-a1da-5769e5bc15ef": {
    "result": "raw-aes-gcm",
    "response": {
      "kind": "configDek",
      "name": "3f64f34c-a0bd-4248-a1da-5769e5bc15ef",
      "key": "5YHAbvomWQuMZSnwQcSWhh4c+XpiildvYlOpbg7VLLE=",
      "encryptionKeyId": "encryptionService",
      "creationTime": "2026-06-30T11:32:03Z",
      "imported": false
    }
  }
}
----

The following command uses a JSON keystore containing the key ID and value obtained in the previous command to decrypt and view a file.

.Command
[source,sh]
----
cbcat --with-keystore '{
  "cipher": "AES-256-GCM",
  "id": "3f64f34c-a0bd-4248-a1da-5769e5bc15ef",
  "key": "5YHAbvomWQuMZSnwQcSWhh4c+XpiildvYlOpbg7VLLE="
}' memcached.rbac
----
====

> == Environment Variables

Reviewer 2: Can we move this section up so it comes right after **Schema** and before **Output**?

Author: That would make sense to me also, but the current location is consistent with the location of this section in all the other command line docs, so I'm going to leave it where it is 🙂

For examples, see [cbbackupmgr backup](https://docs.couchbase.com/server/current/backup-restore/cbbackupmgr-backup.html) or [cbdatarecovery](https://docs.couchbase.com/server/current/tools/cbdatarecovery.html)
